decidim-admin/app/jobs/decidim/admin/newsletter_job.rb

```diff
@@ -8,25 +8,29 @@ class NewsletterJob < ApplicationJob
       queue_as :newsletter
 
       def perform(newsletter)
-        newsletter.with_lock do
-          raise "Newsletter already sent" if newsletter.sent?
+        @newsletter = newsletter
 
-          newsletter.update_attributes!(
+        @newsletter.with_lock do
+          raise "Newsletter already sent" if @newsletter.sent?
+
+          @newsletter.update_attributes!(
             sent_at: Time.current,
             total_recipients: recipients.count,
             total_deliveries: 0
           )
         end
 
         recipients.find_each do |user|
-          NewsletterDeliveryJob.perform_later(user, newsletter)
+          NewsletterDeliveryJob.perform_later(user, @newsletter)
         end
       end
 
       private
 
       def recipients
-        @recipients ||= User.where(newsletter_notifications: true).where.not(email: nil, confirmed_at: nil)
+        @recipients ||= User.where(newsletter_notifications: true, organization: @newsletter.organization)
+                            .where.not(email: nil, confirmed_at: nil)
```

Comment:
`nil` and `""` are different values in the DB, so we might want to check that too. Either that or modify the `NewsletterDeliveryJob` and make it `return unless user.email.present?` (or maybe int he `NewsletterMailer`?)